nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm

```diff
@@ -159,3 +255,139 @@
+	alignment = max(alignment-(rand(0, 4)),0)
+
+
+	// dilithium crystal alignment minigame stolen from ds13
+/obj/machinery/ship_weapon/energy/screwdriver_act(mob/user, obj/item/tool)
+	. = ..()
+	if(maint_state == MSTATE_UNBOLTED)
+		.=TRUE
+		var/sound/thesound = pick('nsv13/sound/effects/computer/beep.ogg','nsv13/sound/effects/computer/beep2.ogg','nsv13/sound/effects/computer/beep3.ogg','nsv13/sound/effects/computer/beep4.ogg','nsv13/sound/effects/computer/beep5.ogg','nsv13/sound/effects/computer/beep6.ogg','nsv13/sound/effects/computer/beep7.ogg','nsv13/sound/effects/computer/beep8.ogg','nsv13/sound/effects/computer/beep9.ogg','nsv13/sound/effects/computer/beep10.ogg','nsv13/sound/effects/computer/beep11.ogg','nsv13/sound/effects/computer/beep12.ogg',)
+		SEND_SOUND(user, thesound)
+		var/list/options = letters
+		for(var/option in options)
+			options[option] = image(icon = 'nsv13/icons/actions/engine_actions.dmi', icon_state = "[option]")
+		var/dowhat = show_radial_menu(user,src,options)
+		if(!dowhat)
+			return
+		combo += "[dowhat]"
+		combocount ++
+		to_chat(user, "<span class='warning'>You inputted [dowhat] into the command sequence.</span>")
+		playsound(src, 'sound/machines/sm/supermatter3.ogg', 20, 1)
+		if(combocount <= 4)
+			addtimer(CALLBACK(src, TYPE_PROC_REF(/atom,screwdriver_act), user),2)   // *scream  addtimer(CALLBACK(object|null, GLOBAL_PROC_REF(type/path|procstring), arg1, arg2, ... argn), time, timertype)
+		if(combocount >= 5) //Completed the sequence
+			if(combo == combo_target)
+				to_chat(user, "<span class='warning'>Realignment of weapon energy direction matrix complete.</span>")
+				playsound(src, 'sound/machines/sm/supermatter1.ogg', 30, 1)
+				freq = max_freq
+				combo = null
+				combocount = 0
+			else
+				to_chat(user, "<span class='warning'>Realignment failed. Continued failure risks dangerous heat overload. Rotating command sequence.</span>")
+				playsound(src, 'nsv13/sound/effects/warpcore/overload.ogg', 100, 1)
+				combo_target = "[pick(letters)][pick(letters)][pick(letters)][pick(letters)][pick(letters)]"
+				heat = max(heat+(heat_per_shot*4),max_heat) //Penalty for fucking it up. You risk destroying the crystal... //well... actually overheating the gun
+				combocount = 0
+				combo = null
+
+/obj/machinery/ship_weapon/energy/multitool_act(mob/living/user, obj/item/multitool/I)
+	if(maint_state == MSTATE_CLOSED)
+		if (istype(I))
+			to_chat(user, "<span class='notice'>You log [src] in the multitool's buffer.</span>")
+			I.buffer = src
+			return TRUE
+	if(maint_state == MSTATE_UNSCREWED)
+		to_chat(user, "<span class='notice'>You must unbolt the protective casing before aligning the lenses!</span>")
+	if(maint_state == MSTATE_UNBOLTED)
+		. = TRUE
+		to_chat(user, "<span class='notice'>You being aligning the lenses.</span>")
+		while(alignment < 100)
+			if(!do_after(user, 5, target = src))
+				return
+			alignment += rand(1,2)
+			if(alignment >= 100)
+				alignment = 100
+				break
+
```

Comment:
```suggestion
	switch(maint_state)
		if(MSTATE_CLOSED)
			if(istype(I))
				to_chat(user, "<span class='notice'>You log [src] in the multitool's buffer.</span>")
				I.buffer = src
				return TRUE
		if(MSTATE_UNSCREWED)
			to_chat(user, "<span class='notice'>You must <I>unbolt</I> the protective casing before aligning the lenses!</span>")
			return FALSE
		if(MSTATE_UNBOLTED)
			to_chat(user, "<span class='notice'>You being aligning the lenses.</span>")
			while(alignment < 100)
				if(!do_after(user, 5, target = src))
					return ..()
				alignment += rand(1,2)
				if(alignment >= 100)
					alignment = 100
					to_chat(user, "<span class='notice'>You finish aligning the lenses.</span>")
					return TRUE
	return ..()
```	
Since it uses states, just a switch is easier. I also cleaned up returns and ..() a bit to make more sense.

Reply:
what does ..() do?